client/src/components/App/Content/SignUp/SignUpForm/useSignUpForm.ts

```diff
@@ -0,0 +1,249 @@
+import { useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import Swal from 'sweetalert2';
+
+import SignUpFields from 'models/enums/SignUpFields';
+import SignUpUser from 'models/SignUpUser';
+import City from 'models/City'
+import County from 'models/County';
+import SourceOrganization from 'models/SourceOrganization'
+import Language from 'models/Language';
+import { Service, Severity } from 'models/Logger';
+import Environment from 'models/enums/Environments';
+import axios from 'Utils/axios'
+import logger from 'logger/logger'
+import StoreStateType from 'redux/storeStateType';
+import { setCities } from 'redux/City/cityActionCreators';
+
+const UserInitialValues: SignUpUser = {
+    [SignUpFields.FULL_NAME] : {
+        [SignUpFields.FIRST_NAME]: undefined,
+        [SignUpFields.LAST_NAME]: undefined,
+    },
+    [SignUpFields.LANGUAGES]: [],
+}
+
+const useSignUp = (props: useSignUpFormInCome) : useSignUpFormOutCome  => {
+
+    const { setCounties, setSourcesOrganization, setLanguages, handleSaveUser } = props;
+
+    const userId = useSelector<StoreStateType, string>(state => state.user.id);
+    const epidemiologyNumber = useSelector<StoreStateType, number>(state => state.investigation.epidemiologyNumber);
+
+    const fetchCities = () => {
+        logger.info({
```

Comment:
should we be logging so much info on the client? 🤷‍♀️ 
we have the same logs on the server, and if the query fails you can still log the error

Reply:
I did the same as roei (rom),
I think we should ask him @RoeiRom